### ISSUE TITLE
Notification problem with the Argos applet

### ISSUE THREAD

**Issue by raphael141714** [Odyseus/CinnamonTools#127]
### Description

It occurs when using the argos cinnamon applet.
Notification displays on startup, saying that the scripts that I feed into argos applet doesn't exists.
But they exists, and are runnable. (chmod 755 ~/path/to/myscript.sh)
Here are my two scripts:

``` bash
#!/usr/bin/env bash

UPTIME_OUTPUT="$(uptime | cut -f1 -d ',' | cut -f4- -d ' ' | tr ':' 'h' | tr 'min' ' ' | tr -d ' ')"
echo "$UPTIME_OUTPUT""m"
```

``` bash
#!/usr/bin/env bash

echo "cpu"
```

They're located in ~/.config/argos/uptime.sh and ~/.config/argos/cpu.sh

### Steps to reproduce

**Actual behavior:**
A strange notification saying "Script file doesn't exist", displaying in the same notification the actual location of the script.
And the script runs nicely in argos, even in multiple instances.

**Reproduces how often:**
Every time when starting Cinnamon.
Also occurs every time I try to make a new argos widget, with a new script.
And also occurs when I change the script used by one of the applets.

### System information

inxi -xxxSc0 -! 31
`System:    Kernel: 4.11.0-1-rt-amd64 x86_64 (64 bit gcc: 6.3.0)
           Desktop: Cinnamon 3.2.7 (Gtk 3.22.16) dm: lightdm Distro: Debian GNU/Linux buster/sid
`

### System logs
[pastebin ~/.cinnamon/glass.log](https://pastebin.com/SrHPFVCX)
[pastebin ~/.xsession-errors](https://pastebin.com/HCHHyZHC)
(Sorry for the french locale)

**Comment by Odyseus** [comment]
Hello, @raphael141714.

I cannot reproduce your issue on any of my systems with different Cinnamon versions (2.8, 3.0, 3.2 and 3.4) and with the latest version of Argos For Cinnamon applet. Using the first script that you posted, it loads perfectly and without any warnings on my systems. Restarting Cinnamon or restarting the system doesn't display the notification either.

Sadly, none of my systems are *pure* Debian systems. They are Linux Mint systems and Archlinux systems. I doubt that it has anything to do with the distribution, but just in case, I'm downloading a live Debian ISO to make more accurate tests.

Looking at your log files, there seems to be no problems generated by the applet logged into them. The `TryExec` warning isn't really a problem because that function is just declared and isn't used (I will comment that function out in the next version, so the warning doesn't show up).

There are a couple of things that I will need you to confirm/check/answer.

1. Make sure that you are using the latest version of the applet (1.02).
2. Is the following message exact error message the notification shows?
    - `Script file doesn't exists. /path/to/script`
    - And how the path is represented? Like this `/path/to/script`? Or like this `file:///path/to/script`?
3. Are you running your system as root? This could screw some calls to the GLib library that check for files existence.
4. Is your HOME partition encrypted? This is just a *shot in the dark*.

**Comment by raphael141714** [comment]
Thanks for replying!
1. The "about" section is saying that I use version 1.2
2. The exact text on the notification is `Script file doesn't exists file:///path/to/the/script.sh`
3. I'm not running as root, but as a standard user with sudo enabled.
4. My HOME isn't encrypted.

I have additionnal information: When using the "Choose script" option from the drop down menu of the applet, the script doesn't produce any notification.
But when I use the "Path to script" option from the configuration window, I get the notification at every cinnamon startup as described earlier.

**Comment by Odyseus** [comment]
Hello, @raphael141714.

> I have additionnal information: ...

That was it. Now I can reproduce it. I never used the settings window to choose the script, I used the **Choose script** context menu item.

This was caused by the path returned by the new Cinnamon settings system implemented since Cinnamon 3.2.x (it returns an URI (file:///path/to/file) instead of a path (/path/to/file)). I thought that I handled that in the code, but I was doing it wrong.

Already fixed on this repository and published on Spices. The main README on this repository has a direct download of the applet package for manual installation in you want to try it immediately.

Thank you very much for your feedback. :+1:
